File: Learning Android App/firebase/alertDialog/app/src/main/java/com/example/alertdialog/MainActivity.java
package com.example.alertdialog;

import androidx.appcompat.app.AppCompatActivity;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                new AlertDialog.Builder(MainActivity.this).setTitle("Exit the Applicatoin")
                        .setMessage("press yes to exit the app").setPositiveButton("Yes", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        finish();
                    }
                }).setNegativeButton("No", new DialogInterface.OnClickListener() {
                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {

                    }
                }).show();


            }
        });
    }

//in general we have to use the alert dialog while we are pressing the back button to exit the application

//    for that we have to override the method name

    @Override
    public void onBackPressed() {
        new AlertDialog.Builder(MainActivity.this).setTitle("Exit the Applicatoin")
                .setMessage("press yes to exit the app").setPositiveButton("Yes", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {
                finish();
            }
        }).setNegativeButton("No", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {

            }
        }).show();

    }
}

File: Learning Android App/retrofit_hack/app/src/main/java/com/example/retrofit_hack/MainActivity.kt
package com.example.retrofit_hack

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val value = Retrofit main = new Retrofit.Builder()
            .baseUrl("https://hackathon-2022-cc.herokuapp.com/")
            .addConverterFactory(GsonConverterFactory.create())
            .build();
    }
}